src/daft-core/src/array/ops/hash.rs

```diff
@@ -122,38 +150,66 @@ fn hash_list(
                 let hashed_child = flat_child
                     .slice(start, end)
                     .ok()?
-                    .hash(Some(&flat_seed))
+                    .hash_with(Some(&flat_seed), hash_function)
                     .ok()?;
                 let child_bytes: Vec<u8> = hashed_child
                     .as_arrow()
                     .values_iter()
                     .flat_map(|v| v.to_le_bytes())
                     .collect();
-                if let Some(cur_seed) = cur_seed_opt {
-                    Some(xxh3_64_with_seed(&child_bytes, cur_seed))
-                } else {
-                    Some(xxh3_64(&child_bytes))
+
+                match hash_function {
+                    HashFunctionKind::XxHash => {
+                        if let Some(cur_seed) = cur_seed_opt {
+                            Some(xxh3_64_with_seed(&child_bytes, cur_seed))
+                        } else {
+                            Some(xxh3_64(&child_bytes))
+                        }
+                    }
+                    HashFunctionKind::MurmurHash3 => {
+                        let hasher = MurBuildHasher::new(cur_seed_opt.unwrap_or(42) as u32);
```

Comment:
Could you please add a comment above why 42 was chosen, other than it being the ultimate answer to life and the universe? Is it possible to use MurBuilderHasher without a seed? I think this is fine to leave as-is, but leave a comment about the "42" being the default seed.